game-core/src/main/java/games/strategy/engine/data/ModifiableProperty.java

```diff
@@ -0,0 +1,145 @@
+package games.strategy.engine.data;
+
+import java.util.function.Supplier;
+
+/**
+ * A wrapper interface to Bundle setters, getters and resetters of the same field.
+ *
+ * @param <T> The type of the field to set, get and reset.
+ */
+public interface ModifiableProperty<T> {
+
+  void setValue(String string) throws Exception;
+
+  void setValue(T value) throws Exception;
+
+  @SuppressWarnings("unchecked")
+  default void setObjectValue(final Object o) {
+    try {
+      if (o instanceof String) {
+        setValue((String) o);
+      } else {
+        setValue((T) o);
+      }
+    } catch (final Exception e) {
+      throw new IllegalStateException("Failed to set Attachment property", e);
+    }
+  }
+
+  T getValue();
+
+  void resetValue();
+
+  /**
+   * Convenience method to create an instance of this interface.
+   */
+  static <T> ModifiableProperty<T> of(
+      final ExceptionConsumer<T> setter,
+      final ExceptionConsumer<String> stringSetter,
+      final Supplier<T> getter,
+      final Runnable resetter) {
+    return new ModifiableProperty<T>() {
```

Comment:
I don't see any other occurrences where a `ModifiableProperty` instance is created.  That suggests this may be a simple value type rather than an interface.